to_json.py

```diff
@@ -0,0 +1,37 @@
+import pandas as pd
+import json
+
+# Load CSV file
+file_path = 'pull_request_comments_commits_codeowners_integrations.csv'
+df = pd.read_csv(file_path)
+
+# Group by Pull Request URL
+grouped_data = []
+for url, group in df.groupby('Pull Request URL'):
+    pull_request_data = {
+        'pull_request_url': url,
+        'comments': []
+    }
+    
+    for _, row in group.iterrows():
+        comment_data = {
+            'comment_type': row['Comment Type'],
+            'user': row['User'],
+            'comment': row['Comment'],
+            'date': row['Date'],
+            'commit_sha': row['Commit SHA'] if pd.notna(row['Commit SHA']) else None,
+            'author': row['Author'],
+            'commit_message': row['Commit Message'] if pd.notna(row['Commit Message']) else None,
+            'is_code_owner': row['Is Code Owner'],
+            'author_is_code_owner': row['Author Is Code Owner']
```

Comment:
I feel like some of this information is redundant- e.g. does it matter to the model for identifying issue what the username, commit_sha, author, commit message is etc? In my repo, I set up something that produced the comment sequence like:

```
(review comment from reviewer) [2023-04-25T07:58:37Z] We can just implement the `native_value` property method instead in this case.
(review comment from reviewer) [2023-04-25T08:30:54Z] If we never use the key in the `NUMBER_TYPES` dict, it doesn't need to be a dict. It could be a tuple, eg.
(review comment from reviewer) [2023-04-25T20:29:26Z] We don't seem to need the check. We can just return the value regardless if it's None or not.
(review comment from author) [2023-04-25T20:35:18Z] 👍 just figured that out on my own. Sorry for the commit spam, currently sitting in the train and the internet isn't fast enough to setup the full development environment on my work laptop... 
(review comment from reviewer) [2023-04-25T20:40:13Z] This is now no longer needed

```suggestion
```
(review comment from reviewer) [2023-04-25T20:40:37Z] What is this call fails? Can it raise? should it be handled?
(review comment from author) [2023-04-25T20:54:11Z] The library could raise a `ValueError` in case the given number is not a multiple of 5. 
If communication with the BMW APIs fail, multiple errors could be raised: `httpx.HTTPStatusError`, `TimeoutError` (if the API cannot reach the vehicle in time) or a plain exception if something totally unexpected happens.

In my mind, we don't need additional error handling in here, as each failure mode is an actual failure we cannot recover from (well, we could retry but that usually doesn't help much - either the API works and it runs through, or there is something wrong and we have to wait for the API/car to recover).
(issue comment from author) [2023-04-25T21:05:07Z] CI fails due to `Error: API rate limit exceeded for installation ID 689304`
(review comment from reviewer) [2023-04-25T21:45:46Z] We do need to handle them. Only ValueError and HomeAssistantError is handled upstream. The rest are bugs and have side effects.



(review comment from author) [2023-04-25T22:16:16Z] But I suppose a broad `except Exception as ex` would not be the way to go? :)

I could implement a custom error class in the library (or rather reuse the ones implemented for #90274 and add more for the remote services in an additional libary version). 

However I am unsure how to best continue with this: Adding error handling should then be done in this PR, added to `select` from #88759 and implemented for [bmw_connected_drive/button.py#L129](https://github.com/home-assistant/core/blob/dev/homeassistant/components/bmw_connected_drive/button.py#L129).

As I won't be able to get something else than `except Exception as ex` through until the beta cut, please advise on how you would like to continue:
- broad exception (and adjust/improve in a later PR)
- keep as is and I'll create another PR in the next days that fixes exception handling for all above (should be able to fix it during beta fix if required)
- don't merge this (and then have a breaking change for 2023.6)
(review comment from author) [2023-04-27T19:03:07Z] Hope this satisfies the requirement (and maybe it can be squeezed into this beta?). Otherwise this will be breaking change for 2022.6 and I'll add the breaking change paragraph again.
(review comment from reviewer) [2023-04-28T12:21:54Z] What will raise a `TypeError`?
(review comment from reviewer) [2023-04-28T12:22:52Z] Please add a link to a changelog or GitHub commit compare view for the version bump in the PR description. Thanks!
(review comment from author) [2023-04-28T12:51:18Z] Forgot that, sorry. It is there now.
(review comment from author) [2023-04-28T12:56:09Z] The climate start/stop services won't trigger `TypeError` right now. When used in notify, it could be raised if parameters are of the wrong type.

If wanted, I can remove it (as it should not occur here).
(review comment from reviewer) [2023-04-28T19:14:19Z] If it can't happen we shouldn't add the code here.
(review comment from author) [2023-04-28T19:15:45Z] Figured as much an already removed in https://github.com/home-assistant/core/pull/91081/commits/b891651f8b1bc830b8a067e0cc26344c80e37df5 
(issue comment from reviewer) [2023-04-28T19:22:32Z] I added it to the milestone to avoid a breaking change.
(issue comment from author) [2023-04-28T19:33:22Z] Just removed the string casting as frenck requested in https://github.com/home-assistant/core/pull/92199#discussion_r1180734104.
Otherwise no changes.
```

which I thought was a good compression of the information for the model.